src/main/webapp/app/shared/import/users-import-dialog.component.ts

```diff
@@ -0,0 +1,267 @@
+import { HttpResponse } from '@angular/common/http';
+import { Subject } from 'rxjs';
+import { ActionType } from 'app/shared/delete-dialog/delete-dialog.model';
+import { CourseManagementService } from 'app/course/manage/course-management.service';
+import { TranslateService } from '@ngx-translate/core';
+import { Exam } from 'app/entities/exam.model';
+import { ExamManagementService } from 'app/exam/manage/exam-management.service';
+import { StudentDTO } from 'app/entities/student-dto.model';
+import { parse } from 'papaparse';
+
+const csvColumns = Object.freeze({
+    registrationNumber: 'registrationnumber',
+    matrikelNummer: 'matrikelnummer',
+    matriculationNumber: 'matriculationnumber',
+    firstNameOfUser: 'firstname',
+    familyNameOfUser: 'familyname',
+    firstName: 'firstname',
+    familyName: 'familyname',
+    lastName: 'lastname',
+    login: 'login',
+    username: 'username',
+    user: 'user',
+    benutzer: 'benutzer',
+    benutzerName: 'benutzername',
+});
+
+type CsvUser = object;
+
+@Component({
+    selector: 'jhi-users-import-dialog',
+    templateUrl: './users-import-dialog.component.html',
+    styleUrls: ['./users-import-dialog.component.scss'],
+    encapsulation: ViewEncapsulation.None,
+})
+export class UsersImportDialogComponent implements OnDestroy {
+    readonly ActionType = ActionType;
+
+    @ViewChild('importForm', { static: false }) importForm: NgForm;
+
+    @Input() courseId: number;
+    @Input() courseGroup: String;
+    @Input() exam: Exam | undefined;
+
+    usersToImport: StudentDTO[] = [];
+    notFoundUsers: StudentDTO[] = [];
+
+    isParsing = false;
+    validationError?: string;
+    isImporting = false;
+    hasImported = false;
+
+    private dialogErrorSource = new Subject<string>();
+    dialogError$ = this.dialogErrorSource.asObservable();
+
+    constructor(
+        private activeModal: NgbActiveModal,
+        private alertService: AlertService,
+        private examManagementService: ExamManagementService,
+        private courseManagementService: CourseManagementService,
+        private translateService: TranslateService,
+    ) {}
+
+    ngOnDestroy(): void {
+        this.dialogErrorSource.unsubscribe();
+    }
+
+    private resetDialog() {
+        this.usersToImport = [];
+        this.notFoundUsers = [];
+        this.hasImported = false;
+    }
+
+    async onCSVFileSelect(event: any) {
+        if (event.target.files.length > 0) {
+            this.resetDialog();
+            this.usersToImport = await this.readUsersFromCSVFile(event, event.target.files[0]);
+        }
+    }
+
+    /**
+     * Reads users from a csv file into a list of StudentDTOs
+     * The column "registrationNumber" is mandatory since the import requires it as an identifier
+     * @param event File change event from the HTML input of type file
+     * @param csvFile File that contains one user per row and has at least the columns specified in csvColumns
```

Comment:
[nit] do we also need to mention the `@return` value?

Reply:
fixed